### ISSUE TITLE
Sprite prefix not getting added to the fragment identifier correctly.

### ISSUE THREAD

**Issue by SowmediaNL** [cascornelissen/svg-spritemap-webpack-plugin#100]
Introduced in v3.0.1: 
https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v3.0.1

This fix breaks the current new ^3.X.X releases, sicne it now outputs double prefixed classes in CSS.
`sprite.svg#myprefixtest-myprefixtest-closure-view`

While the ID's in the SVG remain `myprefixtest-closure-view`

I tested this by editing `css.js` (by undoing this change) and that fixed it.
https://github.com/cascornelissen/svg-spritemap-webpack-plugin/commit/ecb233cc372830d5792b67db49ffd3a34bd3bd94

**Comment by cascornelissen** [comment]
Would you be willing to submit a PR for this? And do all the tests still pass when making this change?

**Comment by SowmediaNL** [comment]
Thanks, 2 pull requests, 1 bugfix and 1 new feature:
https://github.com/cascornelissen/svg-spritemap-webpack-plugin/pulls

**Comment by cascornelissen** [comment]
@SowmediaNL, can you provide a minimal repository with a reproduction. I've been trying to combine options to get this to break but whatever I try the output seems correct to me. 

Especially [this comment](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/pull/101#issuecomment-582793209) on #101 where you state it doesn't occur in the SCSS variant seems weird to me as all formatters (should) have the same logic.

**Comment by fruchtose** [comment]
@cascornelissen I've run into the same issue. I can understand why you would be confused, but I've done some digging which should help.

> Especially this comment on #101 where you state it doesn't occur in the SCSS variant seems weird to me as all formatters (should) have the same logic.

This is not the case. The SCSS variant includes special logic which strips out the prefix from the URL, as seen here: https://github.com/cascornelissen/svg-spritemap-webpack-plugin/blob/42031ca22d8f642e714451a2515c01aa4378a57a/lib/style-formatters/scss.js#L59

This logic is not present in the CSS or LESS output strategies to create the ID fragment in the URL.

**Comment by cascornelissen** [comment]
@fruchtose, I guess you are right 🤔 Thanks for digging into this a little further, do you have time to submit a PR that fixes this for the CSS and LESS style formatters? Simply adding `.replace(options.prefix, '')` should do it, I guess but it'd be nice if you could test if it actually solves the problem on your end.

**Comment by fruchtose** [comment]
@cascornelissen I'll see if I can make time this week. I'm a bit swamped, but I'd love to help. I'll think a bit more about testing—maybe using `Symbol` instances for prefixes and selector names would be viable.

**Comment by cascornelissen** [comment]
I have just checked this a third time but I really can't seem to find the issue. Can someone provide a minimal plugin configuration that shows the issue at hand.

For example (based on the `simple` example):

```
new SVGSpritemapPlugin('src/**/*.svg', {
    styles: path.join(__dirname, 'src/sprites.css'),
    sprite: {
        prefix: 'test-prefix-'
    }
})
```

Generates the following styles, which seems to be correct.

```
.test-prefix-1 { background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3e%3cpath d='M19 19H5V5h14m0-2H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2m-7 14h2V7h-4v2h2'/%3e%3c/svg%3e"); }
.test-prefix-2 { background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3e%3cpath d='M15 15h-4v-2h2a2 2 0 002-2V9a2 2 0 00-2-2H9v2h4v2h-2a2 2 0 00-2 2v4h6m4 2H5V5h14m0-2H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2z'/%3e%3c/svg%3e"); }
.test-prefix-3 { background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3e%3cpath d='M15 15v-1.5a1.5 1.5 0 00-1.5-1.5 1.5 1.5 0 001.5-1.5V9a2 2 0 00-2-2H9v2h4v2h-2v2h2v2H9v2h4a2 2 0 002-2m4 4H5V5h14m0-2H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2z'/%3e%3c/svg%3e"); }
```

ping @fruchtose, @SowmediaNL.

**Comment by cascornelissen** [comment]
Alright, took a fourth time digging into this and I'm pretty sure I've found the issue and fixed it in `master`. @fruchtose or @SowmediaNL, can you confirm it's fixed by installing the dependency directly from GitHub by executing the following command?

```
$ npm install --no-save cascornelissen/svg-spritemap-webpack-plugin#master
```

**Comment by cascornelissen** [comment]
ping @fruchtose and @SowmediaNL, if I get not replies in the next couple of days I guess I'll create a release and close this issue.

**Comment by cascornelissen** [comment]
Forgot to update the issue but this was fixed in [`3.5.6`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v3.5.6) 🚀
